sbws/util/config.py

```diff
@@ -29,52 +39,57 @@ def _read_config_file(conf, fname):
     return conf
 
 
+def _extend_config(conf, fpath):
+    """Extend ConfigParser from file configuration."""
+    conf = _read_config_file(conf, fpath)
+    return conf
+
+
 def _get_default_config():
+    """Return ConfigParser with default configuration."""
     conf = ConfigParser(interpolation=ExtendedInterpolation())
-    fname = os.path.join(PKG_DIR, 'config.default.ini')
-    assert os.path.isfile(fname)
-    conf = _read_config_file(conf, fname)
-    return conf
+    return _read_config_file(conf, DEFAULT_CONFIG_PATH)
 
 
 def _get_user_config(args, conf=None):
+    """Get user configuration.
+    If a custom path is passed, search for user configuration there.
+    In any case, create a minimal user configuration in user configuration path
+    , in case it needs to be overwritten.
+    """
     if not conf:
         conf = ConfigParser(interpolation=ExtendedInterpolation())
     else:
         assert isinstance(conf, ConfigParser)
-    fname = os.path.join(args.directory, 'config.ini')
-    if not os.path.isfile(fname):
-        return conf
-    conf = _read_config_file(conf, fname)
-    return conf
-
-
-def _get_default_logging_config(args, conf=None):
+    if args.config:
+        if not os.path.isfile(args.config):
+            log.warning('Configuration file %s not found, '
+                        'using default configuration, and creating a minimal '
+                        'configuration in %s.', args.config, args.config)
+            _create_user_config_file(args.config)
+            return conf
+        return _extend_config(conf, args.config)
+    _create_user_config_file(USER_CONFIG_PATH)
```

Comment:
If we make `_create_user_config_file()` assert on the file not existing, then we should only conditionally call this function here.